docs(DST-1383, DST-1384): add the Bulk Actions pattern and working example (#5640)
marigold-ui/marigold: docs/content/patterns/user-input/bulk-actions/index.mdx

---
title: Bulk Actions
description: Let users select many records and act on all of them at once.

> badge: alpha

Reviewer: 💡 **`badge: alpha` — still intentional?**

DST-1383 notes the badge "promotes to stable once the working example lands," and the example lands in this same PR. Keeping alpha for bake-in time is reasonable — just confirming it's a choice, not an oversight.

Author: Good catch — it was a choice (bake-in time), but since the example lands in this PR I've promoted it to `beta` here.

| Execution     | waits for the operation to finish      | [`<Button>`](/components/actions/button) `loading` or an inline `<ProgressCircle>` |
| Result        | learns what happened                   | [`<Toast>`](/components/overlay/toast) plus the retained selection on failures     |

## Choosing actions

Not every operation earns a place in the action bar. Offer bulk actions for tasks users genuinely apply to many records at once, not speculatively for every operation a record supports. A bar crowded with rarely used actions slows down the frequent ones.

Name every action with a verb, or a verb plus noun where the target is ambiguous, so its effect is obvious before pressing: "Publish", "Assign category", "Delete". Avoid vague labels such as "Apply" or "OK" on their own. Order the actions from safest to most consequential and place a destructive action such as delete last, visually separated from the routine ones. When a selection carries more actions than fit comfortably, move the long tail into an [`<ActionMenu>`](/components/overlay/action-menu) rather than crowding the bar.

> While the bar is visible, the selection is the object the user works on. Avoid competing per-row actions during that time: a row-level delete next to a bulk delete makes it unclear which scope a press affects. Bulk and per-row actions can still share a table — one-off operations belong in a row menu (see [Table Records](/patterns/user-input/table-record-management)), repetitive ones in the bar — as long as they take turns: while rows are selected, disable the row actions so the bar holds the only pressable scope.

Reviewer: 💡 **Em dashes — the other pattern docs contain none**

```suggestion
While the bar is visible, the selection is the object the user works on. Avoid competing per-row actions during that time: a row-level delete next to a bulk delete makes it unclear which scope a press affects. Bulk and per-row actions can still share a table when they take turns: one-off operations belong in a row menu (see [Table Records](/patterns/user-input/table-record-management)), repetitive ones in the bar, and while rows are selected the row actions disable so the bar holds the only pressable scope.
```

Author: Applied your reword.
